### DISCUSSION TITLE
Two Alertdialogs

### DISCUSSION THREAD

**Discussion by R8bert** [flet-dev/flet#1942] [Q&A] [answered]
### Question

Hi. I have a problem with closing a dialog and open another in the same function

example:

```
    def open_dlg_modalXX1(self, e):
        self.page.dialog = self.dlg_modalXX1
        self.dlg_modalXX1.open = True
        self.page.update()
        time.sleep(20)
        self.dlg_modalXX1.open = False
        self.page.dialog = self.qr_dialog
        self.qr_dialog.open = True
        self.page.update()
```

after i close the 2nd one i have somthing left on my screen
**
![image](https://github.com/flet-dev/flet/assets/61593351/51fd95f3-fafc-4f0d-bb78-110e0b8e82be)
**

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Try adding a small time delay directly after closing the first dialog.

**Reply by R8bert** [reply]
i did it like that 
```
    def open_dlg_modalXX1(self, e):
        self.page.dialog = self.dlg_modalXX1
        self.dlg_modalXX1.open = True
        self.page.update()
        time.sleep(2)
        self.dlg_modalXX1.open = False
        time.sleep(2)
        self.page.dialog = self.qr_dialog
        self.qr_dialog.open = True
        self.page.update()
```
but isnt working, i have the same problem 
maybe is related to `self.page.dialog = self.qr_dialog` in the same function

**Reply by ndonkoHenri** [reply, marked as answer]
After closing the first dlg, update the page and then delay and let's see.

**Reply by R8bert** [reply]
i used `self.dlg_modalXX1.update()` and a time.sleep(0.1) after it
Thanks

**Reply by ndonkoHenri** [reply]
It works now? Or you are still having the bug

**Reply by R8bert** [reply]
yes it works! Thanks
